src/plugins/data/public/index_patterns/expressions/load_index_pattern.ts

```diff
@@ -39,3 +42,17 @@ export function getIndexPatternLoad({
     },
   });
 }
+
+/** @internal */
+export function getIndexPatternLoad({
```

Comment:
whats the reason behind this ?

Reply:
This can probably be cleaned up; the separation between the `get` and `create` functions was a byproduct of how I originally structured this PR before refactoring. Originally I was wiring up the `getStartDependencies` stuff in the `plugin.ts`, but it was adding a bunch of domain-specific logic at the top-level so I relocated it here.

However I never combined the two functions so agree they could be cleaned up.

Although as I think about it, I see pros and cons to the current approach. The main downside is that we have an additional level of indirection that makes things harder to read.

But on the upside, it makes testing slightly easier since you are passing in `getStartDependencies`, which allows you to just mock that one function and run all of your unit tests against `createIndexPatternLoad` and `createEsaggs`. If you handled this logic internally, it would be slightly less convenient to test -- though I suppose if you extracted the logic into a separate internal function you could still mock it away in a unit test.

A simpler alternative implementation might look something like this -- thoughts?

```ts
/** @internal */
async function getScopedDependencies({
  getStartServices,
  kibanaRequest,
}: {
  getStartServices: StartServicesAccessor<DataPluginStartDependencies, DataPluginStart>;
  kibanaRequest: KibanaRequest;
}) {
  const [{ elasticsearch, savedObjects }, , { indexPatterns }] = await getStartServices();
  return {
    indexPatterns: await indexPatterns.indexPatternsServiceFactory(
      savedObjects.getScopedClient(request),
      elasticsearch.client.asScoped(request).asCurrentUser
    ),
  };
}

/** @internal */
export function createIndexPatternLoad({
  getStartServices,
}: {
  getStartServices: StartServicesAccessor<DataPluginStartDependencies, DataPluginStart>;
}) {
  return (): IndexPatternLoadExpressionFunctionDefinition => ({
    ...getIndexPatternLoadMeta(),
    async fn(input, args, { getKibanaRequest }) {
      const kibanaRequest = getKibanaRequest();
      if (!kibanaRequest) {
        throw new Error(
          i18n.translate('data.indexPatterns.indexPatternLoad.error.kibanaRequest', {
            defaultMessage:
              'A KibanaRequest is required to execute this search on the server. ' +
              'Please provide a request object to the expression execution params.',
          })
        );
      }

      const { indexPatterns } = await getScopedDependencies({ getStartServices, kibanaRequest });

      const indexPattern = await indexPatterns.get(args.id);

      return { type: 'index_pattern', value: indexPattern.toSpec() };
    },
  });
}
```

Reply:
discussed offline & agreed to stick to the original implementation. i added some clarifying comments to make the intent more obvious